NuRadioReco/modules/channelGenericNoiseAdder.py

```diff
@@ -145,14 +180,34 @@ def bandlimited_noise(self, min_freq, max_freq, n_samples, sampling_rate, amplit
         elif type == 'rayleigh':
             fsigma = amplitude * sigscale / np.sqrt(2.)
             ampl[selection] = self.__random_generator.rayleigh(fsigma, nbinsactive)
+        elif type == "data-driven":
+            if station_id is None or channel_id is None:
+                self.logger.error("When selecting data-driven noise, the station and channel ids should be passed to bandlimeted noise")
+                raise ValueError
+            
+            scale_parameter_path = f"thermal_noise_scale_parameters_s{station_id}_season23.json"
+            if scale_parameter_path in self.scale_parameter_paths:
```

Comment:
if `scale_parameter_dir` was not passed in the begin function this line will through an error because `self. scale_parameter_paths` is not initalized (correct?).